src/benchmark_register.cc

```diff
@@ -458,4 +458,20 @@ void ClearRegisteredBenchmarks() {
   internal::BenchmarkFamilies::GetInstance()->ClearBenchmarks();
 }
 
+std::vector<int64_t> CreateRange(int64_t lo, int64_t hi, int64_t multi) {
+  std::vector<int64_t> args;
+  internal::AddRange(&args, lo, hi, multi);
+  return args;
+}
+
+std::vector<int64_t> CreateDenseRange(int64_t start, int64_t limit,
+                                      int64_t step) {
+  CHECK_LE(start, limit);
+  std::vector<int64_t> args;
+  for (int64_t arg = start; arg <= limit; arg += step) {
+    args.push_back({arg});
+  }
+  return args;
+}
```

Comment:
can `DenseRange` be rewritten to call this directly?

Reply:
Yes it's possible, but this might not make the code more DRY.
This `CreateRange` returns a vector<int64> while `DenseRange() generates vector<vector<int>>`.
To do this a loop of vector.push_back is still need to convert things like {0, 1, 2, 3} to {{0}, {1}, {2}, {3}}, which is nearly the same as the original code.

Reply:
This will become something like:
```
Benchmark* Benchmark::DenseRange(int64_t start, int64_t limit, int step) {
  CHECK(ArgsCnt() == -1 || ArgsCnt() == 1);
  for (int64_t arg : CreateDenseRange(start, limit, step)) {
    args_.push_back({arg});
  }
  return this;
}
```
Nearly the same as the original, but a temp vector is added.